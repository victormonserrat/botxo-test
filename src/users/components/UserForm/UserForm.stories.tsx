import { action } from "@storybook/addon-actions";
import faker from "faker";
import React from "react";
import UserForm from "./UserForm";

export default {
  title: "User Form",
  component: UserForm,
};

export const Default = () => (
  <UserForm
    firstName={""}
    setFirstName={action("changed first name")}
    lastName={""}
    setLastName={action("changed last name")}
    age={0}
    setAge={action("changed age")}
    connections={0}
    setConnections={action("changed connections")}
    description={""}
    setDescription={action("changed description")}
  />
);

export const Completed = () => (
  <UserForm
    firstName={faker.name.firstName()}
    setFirstName={action("changed first name")}
    lastName={faker.name.lastName()}
    setLastName={action("changed last name")}
    age={faker.random.number(99)}
    setAge={action("changed age")}
    connections={faker.random.number(999)}
    setConnections={action("changed connections")}
    description={faker.lorem.sentences()}
    setDescription={action("changed description")}
  />
);
