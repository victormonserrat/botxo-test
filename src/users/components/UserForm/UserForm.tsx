import React from "react";
import { Form } from "semantic-ui-react";

export interface Props {
  firstName: string;
  setFirstName: (firstName: string) => void;
  lastName: string;
  setLastName: (lastName: string) => void;
  age: number;
  setAge: (age: number) => void;
  connections: number;
  setConnections: (connections: number) => void;
  description: string;
  setDescription: (description: string) => void;
}

export const UserForm: React.FunctionComponent<Props> = ({
  firstName,
  setFirstName,
  lastName,
  setLastName,
  age,
  setAge,
  connections,
  setConnections,
  description,
  setDescription,
}) => {
  return (
    <Form>
      <Form.Group widths="equal">
        <Form.Input
          fluid
          label="First name"
          placeholder="First name"
          value={firstName}
          onChange={(event, { value }) => {
            setFirstName(value);
          }}
        />
        <Form.Input
          fluid
          label="Last name"
          placeholder="Last name"
          value={lastName}
          onChange={(event, { value }) => {
            setLastName(value);
          }}
        />
      </Form.Group>
      <Form.Group widths="equal">
        <Form.Input
          fluid
          label="Age"
          type="number"
          min={0}
          placeholder="Age"
          value={age}
          onChange={(event, { value }) => {
            setAge(+value);
          }}
        />
        <Form.Input
          fluid
          label="Connections"
          type="number"
          min={0}
          placeholder="Connections"
          value={connections}
          onChange={(event, { value }) => {
            setConnections(+value);
          }}
        />
      </Form.Group>
      <Form.TextArea
        label="Description"
        placeholder="Description"
        value={description}
        onChange={(event, { value }) => {
          setDescription(value ? value.toString() : "");
        }}
      />
    </Form>
  );
};

export default UserForm;
