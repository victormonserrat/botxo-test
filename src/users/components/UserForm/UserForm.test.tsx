import { shallow } from "enzyme";
import React from "react";
import UserForm from "./UserForm";

describe("UserForm", () => {
  it("should be rendered correctly", () => {
    const setFirstNameMock = jest.fn();
    const setLastNameMock = jest.fn();
    const setAge = jest.fn();
    const setDescription = jest.fn();
    const setConnections = jest.fn();
    const component = shallow(
      <UserForm
        firstName=""
        setFirstName={setFirstNameMock}
        lastName=""
        setLastName={setLastNameMock}
        age={0}
        setAge={setAge}
        description=""
        setDescription={setDescription}
        connections={0}
        setConnections={setConnections}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
