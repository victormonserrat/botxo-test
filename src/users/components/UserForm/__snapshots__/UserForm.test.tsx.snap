// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserForm should be rendered correctly 1`] = `ShallowWrapper {}`;
